### PR TITLE
Added the ability to set headers in the requests, and add the auth token...

### PR DESCRIPTION
**Pull request by jonnysamps**
... when using the notification service

### CODE REVIEW

**Comment by jonnysamps** [comment]
@vgarro please review

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gemspec
+# gemspec # This causes my `bundle install` to fail
```
Comment (steveburkett) [inline Gemfile:3]: http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/ says to use gemspec

File: lib/services/http_service.rb [hunk 2]
```diff
@@ -19,22 +19,33 @@ def initialize(host)
 
     # @param endpoint     - e.g. '/context'
     # @param query_params - e.g. {a:1,b:2}
-    def get(endpoint = '/', query_params = nil)
-      make_request :get, endpoint, query_params
+    DEFAULT_GET_PARAMS = { endpoint: '/', query_params: nil, body: nil, headers: {} }
```
Comment (vgarro) [inline lib/services/http_service.rb:22]: maybe this tree lines can be:

``` ruby
DEFAULT_GET_PARAMS = { endpoint: '/', query_params: nil, body: nil, headers: {} }
DEFAULT_POST_PARAMS = { endpoint:'/', query_params:nil, body:{}, headers:{} }
DEFAULT_PARAMS = { type: :get, endpoint: '/', query_params: nil, body: nil, headers: {} }
```

``` ruby
DEFAULT_PARAMS = { endpoint: '/', query_params: nil, body: nil, headers: {} }.freeze
DEFAULT_GET_PARAMS = DEFAULT_PARAMS.merge(type: :get)
DEFAULT_POST_PARAMS = DEFAULT_PARAMS.merge(type: :post)
```

File: lib/services/notifications_service.rb [hunk 3]
```diff
@@ -8,18 +8,26 @@ def initialize(http_service)
...
+      auth_token = params[:auth_token] || nil
 
-      uri = "/clients/#{client_urn}/notifications?page=#{page}&size=#{page_size}"
-      response = @http_service.get(uri)
-      get_api_response response.body['notifications'], response.body['total_rows'], response.code, response.message
+
+      uri = "/clients/#{client_urn}/notifications"
+      response = @http_service.get(endpoint: uri,
+                                   query_params: {page: page, size: page_size},
+                                   headers: {'Authorization'=>"Bearer #{auth_token}"})
```
Comment (vgarro) [inline lib/services/notifications_service.rb:20]: maybe this headers could also be a constant

File: lib/services/http_service.rb (hunk 2 above)
Comment (steveburkett) [inline lib/services/http_service.rb:22]: i've never seen constants put just above the methods.  i understand the reasoning, just never seen it :)

File: lib/services/http_service.rb [hunk 4]
```diff
@@ -42,6 +53,9 @@ def make_request(type = :get, endpoint = '/', query_params = nil, body = nil)
 
       http = Net::HTTP.new(uri.host, uri.port)
 
+      # http.set_debug_output $stderr
```
Comment (steveburkett) [inline lib/services/http_service.rb:56]: as a practice, we generally dont leave commented out code in the code + blank lines like this.

File: lib/services/http_service.rb [hunk 5]
```diff
@@ -58,6 +72,10 @@ def make_request(type = :get, endpoint = '/', query_params = nil, body = nil)
         request[CONTENT_TYPE] = APPLICATION_JSON
       end
 
+      if headers
+        headers.each { |key,value| request[key] = value }
```
Comment (steveburkett) [inline lib/services/http_service.rb:76]: headers.each { |key,value| request[key] = value } if headers.present?  is a more concise, ruby like way to do this.  if the line is long, i generally dont use suffix expressions. if short, i do .  this is in the middle in think :)  so, whatever.

File: lib/services/notifications_service.rb [hunk 6]
```diff
@@ -8,18 +8,26 @@ def initialize(http_service)
       @http_service = http_service
     end
 
-    def all(client_urn, params={page: 0, page_size: 25})
-      page      = params[:page]       || 0
-      page_size = params[:page_size]  || 25
+    def all(client_urn, params={page: 0, page_size: 25, auth_token: nil})
+      page       = params[:page]       || 0
```
Comment (steveburkett) [inline lib/services/notifications_service.rb:12]: lining up the || is another thing you dont usually see in ruby code.  i saw this practice back in the .Net days.

File: spec/services/notification_service_spec.rb [hunk 7]
```diff
@@ -52,11 +59,11 @@
 
     let(:notification) do
       G5HubApi::Notification.new('product'=> 'blah',
-                       'locations'=> [],
-                       'notif_type' => 'Some notification',
-                       'description' => 'Bla blab albnskdj wlkjdkjd',
-                       'actions' => [],
-                       'client_id' => '1234')
+                                 'locations'=> [],
```
Comment (steveburkett) [inline spec/services/notification_service_spec.rb:62]: is this a tab?

**Pull request closed without merging**